test/inverse_deconv.cpp

```diff
@@ -72,13 +72,14 @@ void imageTest(string pTestFile, const float gamma, const af_inverse_deconv_algo
 
         ASSERT_EQ(AF_SUCCESS, af_gaussian_kernel(&kerArray, 13, 13, 2.25, 2.25));
 
+        af_dtype itype = (af_dtype)af::dtype_traits<T>::af_type;
         af_dtype otype = (af_dtype)af::dtype_traits<OutType>::af_type;
 
         ASSERT_EQ(AF_SUCCESS, af_load_image(&_inArray, inFiles[testId].c_str(), isColor));
-        ASSERT_EQ(AF_SUCCESS, conv_image<T>(&inArray, _inArray));
+        ASSERT_EQ(AF_SUCCESS, af_cast(&inArray, _inArray, itype));
```

Comment:
it was better earlier because it wasn't using arrayfire functions. you want to test only one function. Otherwise its hard to pinpoint what is failing by looking at the failures.